go/controller/internal/httpserver/handlers/invoke.go

```diff
@@ -0,0 +1,166 @@
+package handlers
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/gorilla/mux"
+	autogen_client "github.com/kagent-dev/kagent/go/autogen/client"
+	"github.com/kagent-dev/kagent/go/controller/internal/httpserver/errors"
+	ctrllog "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// mockAutogenClient interface for testing
+type mockAutogenClient interface {
+	CreateSession(*autogen_client.CreateSession) (*autogen_client.Session, error)
+	CreateRun(*autogen_client.CreateRunRequest) (*autogen_client.CreateRunResult, error)
+}
+
+// InvokeHandler handles agent invocation requests
+type InvokeHandler struct {
+	*Base
+	testClient interface{}
+}
+
+// NewInvokeHandler creates a new InvokeHandler
+func NewInvokeHandler(base *Base) *InvokeHandler {
+	return &InvokeHandler{Base: base}
+}
+
+// SetTestClient allows setting a mock client for testing
+func (h *InvokeHandler) SetTestClient(client interface{}) {
+	h.testClient = client
+}
+
+// InvokeRequest represents a request to invoke an agent
+type InvokeRequest struct {
+	Message string                 `json:"message"`
+	Sync    bool                   `json:"sync"`
+	Context map[string]interface{} `json:"context,omitempty"`
+	UserID  string                 `json:"user_id,omitempty"`
+}
+
+// InvokeResponse represents a response from an agent invocation
+type InvokeResponse struct {
+	SessionID   string `json:"sessionId"`
+	Response    string `json:"response,omitempty"`
+	StatusURL   string `json:"statusUrl,omitempty"`
+	Status      string `json:"status"`
+	CompletedAt string `json:"completedAt,omitempty"`
+}
+
+// HandleInvokeAgent handles POST /api/agents/{agentId}/invoke requests
+func (h *InvokeHandler) HandleInvokeAgent(w ErrorResponseWriter, r *http.Request) {
+	log := ctrllog.FromContext(r.Context()).WithName("invoke-handler").WithValues("operation", "invoke")
+
+	// Extract agent ID from path
+	vars := mux.Vars(r)
```

Comment:
you could use the GetPathParam in `helpers.go` to extract the parameter from URL